File: backend/src/ToDoList.Core/Dtos/CreateTodoItemRequest.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace ToDoList.Core.Dtos
{
    public class CreateTodoItemRequest
    {
        [Required]
        [MaxLength(500)]
        public string Title { get; set; } = string.Empty;

        [MaxLength(10000)]
        public string? Notes { get; set; }

        public long? CategoryId { get; set; }

        public DateTime? DueDate { get; set; }

        public int SortOrder { get; set; }
    }
}

File: clients/android/app/src/main/java/com/yeplist/app/data/remote/dto/CreateTodoItemRequest.kt
package com.yeplist.app.data.remote.dto

data class CreateTodoItemRequest(
    val title: String,
    val notes: String? = null,
    val categoryId: Long? = null,
    val dueDate: String? = null,
    val sortOrder: Int = 0,
    // Real user-edit time (ISO-8601 UTC). Newest-edit-wins conflict arbiter.
    val clientModifiedDate: String? = null
)
